video_xblock/tests/mocks.py

```diff
@@ -0,0 +1,751 @@
+# -*- coding: utf-8 -*-
+"""
+Video XBlock mocks.
+"""
+import json
+from copy import copy, deepcopy
+from mock import Mock
+import requests
+
+from xblock.core import XBlock
+
+from video_xblock.exceptions import VideoXBlockException, VideoXBlockMockException
+from video_xblock.backends import brightcove, youtube, wistia
+
+
+class ResponseStub(object):
+    """
+    Dummy ResponseStub class.
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Delegate kwargs to class properties.
+        """
+        for key, val in kwargs.items():
+            setattr(self, key, val)
+
+    @property
+    def text(self):
+        """
+        Make response compatible with requests.Response.
+        """
+        return getattr(self, 'body', '')
+
+    @property
+    def content(self):
+        """
+        Make response compatible with requests.Response.
+        """
+        return getattr(self, 'body', '')
+
+    def get(self, key):
+        """
+        Allow to fetch data from response body by key.
+        """
+        body = getattr(self, 'body', '')
+        if body:
+            try:
+                return json.loads(body)[key]
+            except KeyError:
+                pass
+
+
+class BaseMock(Mock):
+    """
+    Base custom mock class.
+    """
+
+    # `outcomes` should be in the format of dict().items() to keep the order of items.
+    # First argument: result name, second argument - dictionary containing result data.
+    # Example: (("key1", {}), ("key2", {}), ...)
```

Comment:
Well i used OrderedDict in `ordered_results` but i would like to keep outcomes structure because otherwise we need to add each item separately like
```python
d = OderedDict()
d[key1] = val1
d[key2] = val2
...
d[key_n] = val_n
```